rust/lance/src/dataset/write/commit.rs

```diff
@@ -311,6 +329,15 @@ impl<'a> CommitBuilder<'a> {
                 location: location!(),
             });
         } else {
+            info!(target: TRACE_DATASET_EVENTS, event=CREATE_EVENT, path=&base_path.to_string());
```

Comment:
I think this should be after the commit has succeeded? Otherwise you might record an event that it commits but the commit has actually failed. (Same comment for the other commit traces as well, I think it also means you cannot drive the dataset version as just read version + 1)

Reply:
That's a choice(we trace motivation and result). For rigorousness, it would be better to have `CREATING_EVENT`  and `CREATED_EVENT`. I would change to `CREATED_EVENT`, and move it behind the commit.